src/azure-cli-core/azure/cli/core/commands/query_examples.py

```diff
@@ -0,0 +1,74 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from knack.util import todict
+from knack import events
+from azure.cli.core.commands.events import EVENT_INVOKER_PRE_LOAD_ARGUMENTS
+
+logger = get_logger(__name__)
+
+
+def register_global_query_examples_argument(cli_ctx):
+    '''Register --query-examples argument, and register handler
+    '''
+
+    def handle_example_parameter(cli, **kwargs):  # pylint: disable=unused-argument
+        args = kwargs['args']
+        if args._query_examples is not None:  # pylint: disable=protected-access
+            cli_ctx.invocation.data['output'] = 'table'
+
+            def analyze_output(cli_ctx, **kwargs):
+                tree_builder = TreeBuilder()
+                tree_builder.build(kwargs['event_data']['result'])
+                kwargs['event_data']['result'] = tree_builder.generate_examples(
+                    args._query_examples)  # pylint: disable=protected-access
+                cli_ctx.unregister_event(
+                    events.EVENT_INVOKER_FILTER_RESULT, analyze_output)
+
+            cli_ctx.register_event(
+                events.EVENT_INVOKER_FILTER_RESULT, analyze_output)
+            cli_ctx.invocation.data['query_active'] = True
+
+    def register_query_examples(cli, **kwargs):
+        from knack.experimental import ExperimentalItem
+        commands_loader = kwargs.get('commands_loader')
+        cmd_tbl = commands_loader.command_table
+        experimental_info = ExperimentalItem(cli.local_context.cli_ctx,
+                                             object_type='parameter', target='_query_examples')
+        default_kwargs = {
+            'help': 'Recommend JMESPath string for you. Currently this is a dummy parameter',
+            'arg_group': 'Global',
+            'is_experimental': True,
+            'nargs': '*',
```

Comment:
could we use `+` not `*` here?

Reply:
Because this argument also works when no positional argument is provided.